worker-taskunstowing/src/main/java/com/github/jobservice/workers/taskunstowing/TaskUnstowingWorker.java

```diff
@@ -0,0 +1,195 @@
+                        String.format("Sent unstowed task message with partition id %s, job id %s and job task id %s to queue %s, but "
+                            + "failed to delete unstowed task message from database", partitionId, jobId,
+                                      taskMessage.getTracking().getJobTaskId(), taskMessage.getTo()));
+                    if (DatabaseExceptionChecker.isTransientException(exception)) {
+                        LOGGER.error(failure.getFailureMsg(), exception);
+                        throw new DocumentWorkerTransientException(failure.getFailureMsg(), exception);
+                    } else {
+                        processFailure(failure, exception, document);
+                    }
+                }
+            } catch (final Exception exception) {
+                final TaskUnstowingWorkerFailure failure = new TaskUnstowingWorkerFailure(
+                    TaskUnstowingWorkerFailure.FAILED_TO_SEND_UNSTOWED_TASK_MESSAGE_TO_QUEUE_ID,
+                    String.format("Failed to send unstowed task message with partition id %s, job id %s and job task id %s to queue %s",
+                                  partitionId, jobId, taskMessage.getTracking().getJobTaskId(), taskMessage.getTo()));
+                processFailure(failure, exception, document);
+            }
+        }
+    }
+
+    private static void processFailure(
+        final TaskUnstowingWorkerFailure TaskUnstowingWorkerFailure,
+        final Document document)
+    {
+        LOGGER.error(TaskUnstowingWorkerFailure.getFailureMsg());
+        TaskUnstowingWorkerFailure.addToDoc(document);
+    }
+
+    private static void processFailure(
+        final TaskUnstowingWorkerFailure TaskUnstowingWorkerFailure,
+        final Throwable cause,
+        final Document document)
+    {
+        LOGGER.error(TaskUnstowingWorkerFailure.getFailureMsg(), cause);
+        TaskUnstowingWorkerFailure.addToDoc(document, cause);
+    }
+
+    private static TaskMessage convertStowedTaskRowToTaskMessage(final StowedTaskRow stowedTaskRow) throws IOException
+    {
+        final TrackingInfo trackingInfo = new TrackingInfo(
+            stowedTaskRow.getTrackingInfoJobTaskId(),
+            new Date(stowedTaskRow.getTrackingInfoLastStatusCheckTime()),
+            stowedTaskRow.getTrackingInfoStatusCheckIntervalMillis(),
+            stowedTaskRow.getTrackingInfoStatusCheckUrl(),
+            stowedTaskRow.getTrackingInfoTrackingPipe(),
+            stowedTaskRow.getTrackingInfoTrackTo());
+
+        return new TaskMessage(
+            UUID.randomUUID().toString(),
+            stowedTaskRow.getTaskClassifier(),
+            stowedTaskRow.getTaskApiVersion(),
+            stowedTaskRow.getTaskData(),
+            TaskStatus.valueOf(stowedTaskRow.getTaskStatus()),
+            stowedTaskRow.getContext() != null
+            ? OBJECT_MAPPER.readValue(stowedTaskRow.getContext(), Map.class)
```

Comment:
This deserializes to a `Map<String, byte[]>` correctly even though it was only told to deserialize to a `Map`?

Reply:
Thanks, I've updated this to use a TypeReference now to specify that the type should be `Map<String, byte[]>`

Reply:
I think the problem here is this:

1) A task message is sent to any worker with a context that looks like this:

```
  "context": {
    "caf/worker": "ew0KICAia2V5MSI6ICJ2YWx1ZTEiDQp9"
  },
```

The key must be `caf/worker` otherwise won't get passed to the worker. 

The value is a base64 encoded map:

https://github.com/WorkerFramework/worker-framework/blob/develop/worker-core/src/main/java/com/hpe/caf/worker/core/WorkerTaskImpl.java#L122

2) The worker framework uses the `caf/worker` key to pull the value out of this `Map<String, byte[]>`, passing on just the `byte[]` to the worker:

https://github.com/WorkerFramework/worker-framework/blob/develop/worker-core/src/main/java/com/hpe/caf/worker/core/WorkerTaskImpl.java#L122

3) The Task Stowing Worker stores this `byte[]` in the database:

https://github.com/JobService/worker-taskstowing/blob/develop/worker-taskstowing/src/main/java/com/github/jobservice/workers/taskstowing/TaskStowingWorker.java#L102

4) The Task Unstowing Worker reads this `byte[]` from the database, however, the Task Unstowing Worker actually needs a `Map<String, byte[]>` in order to create a TaskMessage to sent onto the intended worker:

https://github.com/JobService/worker-taskunstowing/blob/SCMOD-12730/worker-taskunstowing/src/main/java/com/github/jobservice/workers/taskunstowing/TaskUnstowingWorker.java#L174

So basically, I think we the Task Unstowing Worker needs to know this `caf/worker` (servicePath) key in order to reconstruct the context for the TaskMessage, I'm just not sure the best way to go about it?

Reply:
Removed context as discussed